wire-ios-sync-engine/Source/Data Model/Conversation+Deletion.swift

```diff
@@ -62,23 +62,26 @@ extension ZMConversation {
 
             if response.httpStatus == 200 {
 
+                var conversation: ZMConversation?
                 contextProvider.syncContext.performGroupedBlock {
+                    conversation = ZMConversation.fetch(
+                        with: conversationId,
+                        domain: nil,
+                        in: contextProvider.syncContext
+                    )
+                }
 
-                    guard let conversation = ZMConversation.fetch(
-                            with: conversationId,
-                            domain: nil,
-                            in: contextProvider.syncContext
-                    ) else {
-                        return
-                    }
+                guard let conversation else {
+                    return completion(.success(()))
+                }
 
-                    removeLocalConversation.invoke(
+                Task {
+                    await removeLocalConversation.invoke(
                         with: conversation,
                         syncContext: contextProvider.syncContext
                     )
+                    completion(.success(()))
```

Comment:
@typfel is the closure called on the correct thread?

Reply:
No, it's not. We want call the completion closure on the main thread.